### PR TITLE
ST-1403-unit tests

### PR DESCRIPTION
**Pull request by geciciserkan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ST-1403

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/app/case/answers/formatDate.test.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  it('returns empty string when date is invalid', () => {
+    const date = {
+      day: '31',
+      month: '02',
+      year: '2023',
+    };
+
+    const formattedDate = getFormattedDate(date);
+    expect(formattedDate).toEqual('');
+  });
```
Comment (osheaf01) [inline src/main/app/case/answers/formatDate.test.ts:45]: Could you add a test for Leap Years, 29th February? and maybe century leap years, though that's unusual...

File: src/main/app/case/formatter/address.test.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    const formattedAddress = getFormattedAddress(data);
+    expect(formattedAddress).toEqual(
+      '123 Main Street<br>Apt 4B<br>Building 2<br>Cityville<br>Countyshire<br>12345<br>Countryland'
+    );
+  });
+
+  it('returns formatted address with line breaks when some address fields are missing', () => {
+    const data = {
+      applicantAddress1: '123 Main Street',
```
Comment (osheaf01) [inline src/main/app/case/formatter/address.test.ts:23]: Are tests for error handling where some critical address fields are missing appropriate? I.e. if there's no address line 1, or no town/city?

File: src/main/app/case/answers/formatDate.test.ts (hunk 1 above)
Comment (geciciserkan) [reply src/main/app/case/answers/formatDate.test.ts:45]: https://github.com/hmcts/sptribs-frontend/pull/363/files I created this PR for that

File: src/main/app/case/formatter/address.test.ts (hunk 2 above)
Comment (geciciserkan) [reply src/main/app/case/formatter/address.test.ts:23]: This address function is not being used. I will leave it as is

**Pull request merged**